plugins/shell/prism-shell.js

```diff
@@ -0,0 +1,69 @@
+(function() {
+
+if (typeof self === 'undefined' || !self.Prism || !self.document) {
+	return;
+}
+
+Prism.hooks.add('complete', function (env) {
+	if (!env.code) {
+		return;
+	}
+
+	// Works only for <kbd> wrapped inside <pre> (not inline).
+	var pre = env.element.parentNode;
+	var clsReg = /\s*\bshell\b\s*/;
+	if (
+		!pre || !/pre/i.test(pre.nodeName) ||
+			// Abort only if neither the <pre> nor the <kbd> have the class
+		(!clsReg.test(pre.className) && !clsReg.test(env.element.className))
+	) {
+		return;
+	}
+
+	if (env.element.querySelector('.shell-prompt')) {
+		// Abort if line numbers already exists
+		return;
+	}
+
+	if (clsReg.test(env.element.className)) {
+		// Remove the class "shell" from the <kbd>
+		env.element.className = env.element.className.replace(clsReg, '');
+	}
+	if (!clsReg.test(pre.className)) {
+		// Add the class "shell" to the <pre>
+		pre.className += ' shell';
+	}
+
+	// Create the "rows" that will become the shell prompts. -- cwells
+	var user = pre.getAttribute('data-user') || '';
+	var host = pre.getAttribute('data-host') || '';
```

Comment:
How about some sensible defaults for `user` and `host` like "user" and "localhost". Do you think that would be useful or maybe just confusing?